integration-tests/src/test/java/org/openmrs/module/fhir2/providers/r3/TaskFhirResourceProviderIntegrationTest.java

```diff
@@ -0,0 +1,423 @@
+		
+	}
+	
+	@Test
+	public void shouldReturnBadRequestWhenDocumentIdDoesNotMatchTaskIdAsXML() throws Exception {
+		// get the existing record
+		MockHttpServletResponse response = get("/Task/" + TASK_UUID).accept(FhirMediaTypes.XML).go();
+		Task task = readResponse(response);
+		
+		// update the existing record
+		task.setId(WRONG_TASK_UUID);
+		
+		// send the update to the server
+		response = put("/Task/" + WRONG_TASK_UUID).xmlContext(toXML(task)).go();
+		assertThat(response, isBadRequest());
+		assertThat(response.getContentType(), is(FhirMediaTypes.XML.toString()));
+		assertThat(response.getContentAsString(), notNullValue());
+		
+		OperationOutcome operationOutcome = readOperationOutcome(response);
+		
+		assertThat(operationOutcome, notNullValue());
+		assertThat(operationOutcome.hasIssue(), is(true));
+		
+	}
+	
+ @Test
+ public void shouldReturnBadRequestWhenDocumentIdDoesNotMatchTaskIdAsJSON() throws Exception {
+  // get the existing record
+  MockHttpServletResponse response = get("/Task/" + TASK_UUID).accept(FhirMediaTypes.JSON).go();
+  Task task = readResponse(response);
+  
+  // update the existing record
+  task.setId(WRONG_TASK_UUID);
+  
+  // send the update to the server
+  response = put("/Task/" + WRONG_TASK_UUID).jsonContent(toJson(task)).jsonContent(toJson(task)).go();
+  assertThat(response, isBadRequest());
+  assertThat(response.getContentType(), is(FhirMediaTypes.XML.toString()));
+  assertThat(response.getContentAsString(), notNullValue());
+  
+  OperationOutcome operationOutcome = readOperationOutcome(response);
+  
+  assertThat(operationOutcome, notNullValue());
+  assertThat(operationOutcome.hasIssue(), is(true));
+ }
+ 
+  
+	@Test
+	public void shouldReturnNotFoundWhenUpdatingNonExistentTaskAsXML() throws Exception {
+		// get the existing record
+		MockHttpServletResponse response = get("/Task/" + TASK_UUID).accept(FhirMediaTypes.XML).go();
+		Task task = readResponse(response);
+		
+		// update the existing record
+		task.setId(WRONG_TASK_UUID);
+		
+		// send the update to the server
+		response = put("/Task/" + TASK_UUID).xmlContext(toXML(task)).go();
+		
+		assertThat(response, isNotFound());
+		assertThat(response.getContentType(), is(FhirMediaTypes.XML.toString()));
+		assertThat(response.getContentAsString(), notNullValue());
+		
+		OperationOutcome operationOutcome = readOperationOutcome(response);
+		
+		assertThat(operationOutcome, notNullValue());
+		assertThat(operationOutcome.hasIssue(), is(true));
+		
+	}
+ @Test
+ public void shouldReturnNotFoundWhenUpdatingNonExistentTaskAsJSON() throws Exception {
+  // get the existing record
+    MockHttpServletResponse response = get("/Task/" + TASK_UUID).accept(FhirMediaTypes.JSON).go();
+    Task task = readResponse(response);
+    
+    // update the existing record
+    task.setId(WRONG_TASK_UUID);
+    
+    // send the update to the server
+    response = put("/Task/" + WRONG_TASK_UUID).jsonContent(toJson(task)).jsonContent(toJson(task)).go();
```

Comment:
```suggestion
    response = put("/Task/" + WRONG_TASK_UUID).jsonContent(toJson(task)).go();
```

Reply:
hey @ibacher  @varung-31 , can you take a look once more thanks